main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/url"
+)
+type (
+	Visitor struct {
+		Host       string
+		UserAgent  string
+		RequestUri string
+		Headers    http.Header
+		QueryParams url.Values
+	}
+)
+
+func main() {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		jsonVisitorData, err := json.MarshalIndent(
+			Visitor{
+				r.Host,
+				r.Header.Get("User-Agent"),
+				r.URL.Path,
+				r.Header,
+				r.URL.Query(),
+			}, "", "    ")
+		if err != nil {
+			panic(err)
```

Comment:
Please do not use panic here. It means that in case of error in process of handling request - whole server will be stopped.
Return an error and error code instead.